src/lib.rs

```diff
@@ -300,6 +316,7 @@ impl fmt::Display for MaildirError {
             Io(ref e) => write!(f, "IO Error: {}", e),
             Utf8(ref e) => write!(f, "UTF8 Encoding Error: {}", e),
             Time(ref e) => write!(f, "Time Error: {}", e),
+            Naming(ref e) => write!(f, "Invalid Folder Name: {}", e),
```

Comment:
Right now this will print out `Invalid Folder Name: Invalid Folder Name` which is not great. Instead if you implement the suggestion above of storing a string, you can have it print out the string `Invalid Folder Name: ...` where `...` would be the actual invalid folder name.